src/systemctl/systemctl.c

```diff
@@ -6327,6 +6328,137 @@ static int log_level(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int service_name_to_dbus(sd_bus *bus, const char *name, char **ret_dbus_name) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_free_ char *bus_name = NULL;
+        _cleanup_strv_free_ char **names = NULL;
+        char *p;
+        int r;
+
+        /* First, look for the BusName= property */
+        _cleanup_free_ char *dbus_path = unit_dbus_path_from_name(name);
+        if (!dbus_path)
+                return log_oom();
+
+        r = sd_bus_get_property_string(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                dbus_path,
+                                "org.freedesktop.systemd1.Service",
+                                "BusName",
+                                &error,
+                                &bus_name);
+        if (r < 0)
+                return log_error_errno(r, "Failed to obtain BusName= property of %s: %s",
+                                       name, bus_error_message(&error, r));
+
+        if (!isempty(bus_name))
+                p = TAKE_PTR(bus_name);
+        else {
+                /* Second, look for dbus-service name among the aliases */
+                r = sd_bus_get_property_strv(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                dbus_path,
+                                "org.freedesktop.systemd1.Unit",
+                                "Names",
+                                &error,
+                                &names);
```

Comment:
I'd really drop the alias thing. we load aliases only if they are requested, and if they are never requested then the `Names` property will not list them. hence it is not a reliable way to figure out bus services. It will work sometimes and sometimes not, depending how the service was first activated. it's a game...

I think we should really make this explicit, i.e. only honour BusName= for this, and add good explanations in the man page in the error message suggesting people to add BusName= to their service if they want this to work. i.e. if you invoke this command and you get back "Your service %s is not a bus service or does not declare BusName= setting. Please add it and implement XYZ". 

I think in the long-run everyone wins if services have BusName= set properly, i.e. carry correct metadata, and this might be the way to get there.

I mean, after all this stuff won't work without explicit support in the service anyway, as the services have to explicitly implement the LogControl1 interface. And if they implement that, then also adding BusName= to the service file is kinda implicit... 

I prepped # now, which adds the missing BusName= to all our own services where it was missing.

So I'd propose:

1. Let's make the man page description of "systemctl service-log-level" explicit, and say that the services need to implement LogControl1 + set BusName= for this to work.
2. Update the BusName= description in the man page, to say that this is useful both for Type=dbus *and* `systemctl service-log-level`
3. Update the org.freedesktop.LogControl1 man page to point to the systemctl verb, and that BusName= should be set
4. Update the systemctl command to give very explicit hints if invoked on services that do not carry BusName= or that do not impement LogControl1, maybe even if an urlified terminal hyperlink to the org.freedesktop.LogControl1 man page.

I think the outcome of this would be best: everything is systematic, easy to discover and well documented